### ISSUE TITLE
Compilation errors for typescript declarations

### ISSUE THREAD

**Issue by goloveychuk** [amcharts/amcharts4#72]
Declarations are failed to compile when used
`"strictFunctionTypes": true,` option in `tsconfig.json`.
As I know it enables bivariance checks for function types

```ts
function a(b: (c: string|number)=> string) {
  b(2)
}
a((b: string)=>b.concat('sd')) // compilation error with strictFnTypes enabled
```

**Comment by martynasma** [comment]
amCharts 4 code is currently not quite strict mode-ready. We are working on little nitbits that are preventing it, but still are a long way from completing it.

We'll keep you posted.

**Comment by martynasma** [comment]
We're still working on full support for TypeScript's strict mode in amCharts 4. If you are using strict mode, you must set these options in your `tsconfig.json`:

```
{
  "compilerOptions": {
    "strictNullChecks": false,
    "strictFunctionTypes": false
  }
}
```

**Comment by teuf22** [comment]
My project uses strictFunctionTypes:true. To include amcharts library I would have to turn off the warning for the entire project so this is non-starter. I found that strictFunctionTypes is useful in finding bugs early. I hope that you can achieve support for strict mode soon.

**Comment by Pauan** [comment]
Thanks to @goloveychuk 's help, we now fully support `strictFunctionTypes`! It will be in the next release.

Note: we still don't support `strictNullChecks`, that is a much bigger change, but we are working on it.

**Comment by Pauan** [comment]
@martynasma Reminder to update the docs to remove all mentions of `strictFunctionTypes` when making the next release (we still need to mention `strictNullChecks`).

**Comment by martynasma** [comment]
Fixed in [4.0.22](https://github.com/amcharts/amcharts4/releases/tag/4.0.22).

## [4.0.22] - 2019-01-23

### Added
- 23 new locales: Arabic, Bosnian, Catalan, Czech, Greek, Estonian, Finnish, Hebrew, Hindi, Croatian, Hungarian, Indonesian, Japanese, Korean, Latvian, Polish, Romanian, Serbian, Thai, Turkish, Vietnamese, Chinese (simplified and traditional versions). Thanks Bjorn Svensson!

### Fixed
- It is no longer necessary to use `"strictFunctionTypes": false`, [thanks to goloveychuk for their help](https://github.com/amcharts/amcharts4/pull/645).
- Broken images were causing export to fail.
- Setting Series' `name` to a number with percent sign was resulting in breakage of the whole chart.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
